streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -1205,6 +1212,20 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
       }
   }
 
+  private final def processDefault: ZManaged[R, E, InputStream[E, A]] =
+    for {
+      queue <- Queue.bounded[Take[E, A]](1).toManaged(_.shutdown)
+      _     <- self.intoManaged(queue).fork
+    } yield queue.take.flatMap {
```

Comment:
There is an existing combinator called toQueue that does this.

Reply:
Thanks. Done.